cmd/ddev/cmd/ssh.go

```diff
@@ -46,7 +53,7 @@ ddev ssh -d /var/www/html`,
 			Cmd:     shell + " -l",
 			Dir:     sshDirArg,
 		})
-		if err != nil {
+		if err != nil && isPiped {
 			util.Failed("Failed to ddev ssh %s: %v", serviceType, err)
 		}
```

Comment:
The easy part of this PR is to remove the output on failure as it's not useful, and in the case of `ddev ssh` was already output if needed, and just exit with the exit code, which can be done something like this:

```suggestion
		if err != nil { // Didn't experiment with your isPiped here...
			if exiterr, ok := err.(*exec.ExitError); ok {
				os.Exit(exiterr.ExitCode())
			} else {
				os.Exit(1)
			}
		}
```

Reply:
Interesting.  So many moving parts on top of just getting my head around Go, so every little suggestion is appreciated.  Will check this out!